### PR TITLE
Adding custom cache key support

### PR DESCRIPTION
**Pull request by adityaj221**
Adding support for creating a custom cache route key as the present system creates a new cache if there are different params added to the URL.

For Ex:

router.get("/index/*", routeCache.cacheSeconds(4536000), (req, res) => });

For the above if I call with "/index?a=1", a new cache is create with the same name.
If I call "/index?a=2", this would create a new cache with the new name.

Instead Added a support for another param as "cacheKey", which can be set same as the route but which would be used as the key to create the cache rather than the originalURL.

### CODE REVIEW

**Comment by fluse** [comment]
pls merge!

**Comment by bradoyler** [comment]
@adityaj221 Sorry for the slow response, i like this idea, would you be able to add a mocha test to show the intended usage? ie. https://github.com/bradoyler/route-cache/blob/master/test/index.js#L14

**Comment by adityaj221** [comment]
This is the error message linked to the last commit, just to demonstrate the issue with the existing code:

# RouteCache middleware test
    ◦ 1st Hello:   ✓ 1st Hello (40ms)
    ◦ 2nd Hello:   ✓ 2nd Hello 
    ◦ 1st Redirect to hello:   ✓ 1st Redirect to hello 
    ◦ 2nd Redirect to hello:   ✓ 2nd Redirect to hello 
    ◦ 301 Redirect to hello:   ✓ 301 Redirect to hello 
    ◦ ~ delayed 301 Redirect to hello:   ✓ ~ delayed 301 Redirect to hello (1202ms)
    ◦ Explicit 302 Redirect to hello:   ✓ Explicit 302 Redirect to hello 
    ◦ ~ delayed: Hello 2:   ✓ ~ delayed: Hello 2 (1204ms)
    ◦ Hello test with param:   1) Hello test with param
    ◦ Error states don't get cached:   ✓ Error states don't get cached 
    ◦ test removeCache:   ✓ test removeCache (1206ms)
    ◦ res.json headers:   ✓ res.json headers (206ms)
    ◦ Params test without params:   ✓ Params test without params 
    ◦ Params test with a = 1:   ✓ Params test with a = 1 
    ◦ Params test with a = 2:   ✓ Params test with a = 2 
  14 passing (4s)
  1 failing
  1) # RouteCache middleware test Hello test with param:
     Error: expected 'Hello 2' response body, got 'Hello 3'
      at error (/home/travis/build/bradoyler/route-cache/node_modules/supertest/lib/test.js:227:13)
      at Test.assert (/home/travis/build/bradoyler/route-cache/node_modules/supertest/lib/test.js:179:21)
      at /home/travis/build/bradoyler/route-cache/node_modules/supertest/lib/test.js:126:10
      at Test.Request.callback (/home/travis/build/bradoyler/route-cache/node_modules/superagent/lib/node/index.js:660:30)
      at Test.<anonymous> (/home/travis/build/bradoyler/route-cache/node_modules/superagent/lib/node/index.js:132:10)
      at emitOne (events.js:95:20)
      at Test.emit (events.js:182:7)
      at IncomingMessage.<anonymous> (/home/travis/build/bradoyler/route-cache/node_modules/superagent/lib/node/index.js:818:12)
      at emitNone (events.js:85:20)
      at IncomingMessage.emit (events.js:179:7)
      at endReadableNT (_stream_readable.js:913:12)
      at process._tickCallback (internal/process/next_tick.js:98:9)


Committing again after removing the error case.

**Comment by bradoyler** [comment]
@adityaj221 would u mind replaying this change against master? I'd like to get this merged.

**Comment by bradoyler** [comment]
we should actually add back the test `'Hello test with param'` to should the change in behavior.
ie.
```
  it('Hello test with param', function (done) {
    agent
      .get('/hello?a=1')
      .expect('Hello 3', done);
  });
```

**Comment by adityaj221** [comment]
I've updated the test case and also have resolved the conflicts.

File: index.js [hunk 1]
```diff
@@ -8,15 +8,14 @@ var cacheStore = new LRU({
...
   var ttl = secondsTTL * 1000
-
   return function (req, res, next) {
-    var key = req.originalUrl
+    var key = (cacheKey === undefined) ? req.originalUrl : cacheKey
     if (redirects[key]) {
       return res.redirect(redirects[key].status, redirects[key].url)
     }
-
+    console.log(key)
```
Comment (bradoyler) [inline index.js]: @adityaj221 please remove

**Pull request merged**